### PR TITLE
Remove unused environment from anomaly detector classes

### PR DESCRIPTION
**Pull request by davidkyle**
Following #54371 I realised the `Environment` variable being passed in to the various AD managers is unused.

This takes removes the environment var and some minor tidying up to clean up Intellij inspection warnings

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java [hunk 1]
```diff
@@ -333,10 +330,16 @@ public void writeUpdateProcessMessage(JobTask jobTask, UpdateParams updateParams
                     updateProcessMessage.setFilter(filter);
 
                     if (updateParams.isUpdateScheduledEvents()) {
-                        jobManager.getJob(jobTask.getJobId(), new ActionListener<Job>() {
+                        jobManager.getJob(jobTask.getJobId(), new ActionListener<>() {
                             @Override
                             public void onResponse(Job job) {
-                                DataCounts dataCounts = getStatistics(jobTask).get().v1();
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java]: Intellij flagged this as a get on an optional without a isPresent check which is a very valid warning. In practice I don't think we could hit an NPE here but it shows the value in the inspections _if you can keep the noise level down_.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java [hunk 2]
```diff
@@ -333,10 +330,16 @@ public void writeUpdateProcessMessage(JobTask jobTask, UpdateParams updateParams
...
+                                Optional<Tuple<DataCounts, Tuple<ModelSizeStats, TimingStats>>> stats = getStatistics(jobTask);
+                                DataCounts dataCounts;
+                                if (stats.isPresent()) {
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java]: nit: make it a single liner using ternary?

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM Nice! Just left an optional suggestion.

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**